### ISSUE TITLE
restify.NotAuthorizedError should be renamed/deprecated

### ISSUE THREAD

**Issue by natesilva** [restify/node-restify#352]
I’d like to suggest `restify.NotAuthorizedError` (403) be renamed or deprecated in favor of `restify.ForbiddenError`.

* In the HTTP standard, 401 is defined as “Unauthorized”.
* In the HTTP standard, 403 is defined as “Forbidden”.

In plain English, “not authorized” means the same thing as “unauthorized”. But 403 !== 401.

I understand that `restify.InvalidCredentialsError` maps to 401, and I don’t think that needs to change.

It’s 403/`NotAuthorizedError` that has potential for confusion. For example, see #181 where even @mcavage confuses `NotAuthorizedError` with 401.

**Comment by micahr** [comment]
403s as NotAuthorized makes more sense in the context of a REST api where typically someone not having the appropriate permissions to a URI should receive instructions of where to achieve the proper authorization, not be Forbidden.

I don't know the history of this change, but the current implementation seems to line up well with the HTTP spec.
